apps/client/src/app/pages/portfolio/analysis/analysis-page.component.ts

```diff
@@ -201,13 +202,42 @@ export class AnalysisPageComponent implements OnDestroy, OnInit {
 
   public onChangeGroupBy(aMode: GroupBy) {
```

Comment:
Can you adapt the (monthly) savings rate according to the mode? (it can be set in the _FIRE_ calculator and need to adapted if the mode is `year`).

I would add a class attribute `public savingsRate: number;` on top and do something like this:

```
this.savingsRate = (hasImpersonationId || user.settings.isRestrictedView) ? undefined : user?.settings?.savingsRate;

if (aMode === 'year') {
  this.savingsRate = 12 * this.savingsRate;
}
```

Update the template accordingly:

```
<gf-investment-chart
  ...
  [savingsRate]="savingsRate">
></gf-investment-chart>
```

Reply:
Besides that, can you update (twice)

```
<gf-investment-chart
  ...
  groupBy="month"
  ...
></gf-investment-chart>
```
to 
```
<gf-investment-chart
  ...
  [groupBy]="mode"
  ...
></gf-investment-chart>
```
I think it will still have the same behaviour 🙂

Reply:
```
this.savingsRate = (hasImpersonationId || user.settings.isRestrictedView) ? undefined : user?.settings?.savingsRate;
```

Created a getter for `savingsRate` as `hasImpersonationId` and `user` are set inside different subscribe blocks

Reply:
Nice!